auth_dynamic_groups_contract/views/res_groups.xml

```diff
@@ -0,0 +1,18 @@
+<odoo>
+
+    <record id="view_groups_form" model="ir.ui.view">
+        <field name="inherit_id" ref="auth_dynamic_groups.view_groups_form" />
+        <field name="model">res.groups</field>
+        <field name="arch" type="xml">
+            <field name="is_dynamic" position="after">
+                <field name="is_contract_based" />
+            </field>
+            <field name="is_dynamic" position="attributes">
+                <attribute
+                    name="attrs"
+                    >{'readonly': [('is_contract_based', '=', True)]}</attribute>
+            </field>
```

Comment:
I'd suggest to make the condition also readonly, we don't want the user to fiddle with the condition then